sys/net/crosslayer/ng_netif/ng_netif_addr_from_str.c

```diff
@@ -14,47 +15,51 @@
 
 #include "net/ng_netif.h"
 
-static uint8_t _parse_byte(char *str)
+static int _dehex(char c)
 {
-    uint8_t res = 0;
-
-    for (int i = 0; i < 2; i++) {   /* iterate over half-bytes */
-        if ((str[i] >= '0') && (str[i] <= '9')) {       /* if '0'-'9' */
-            res |= (str[i] - '0') << (4 * (1 - i));     /* set half-byte to 0-9 */
-        }
-        else if ((str[i] >= 'a') && (str[i] <= 'f')) {  /* if 'a'-'f' */
-            res |= (str[i] - 'a' + 10) << (4 * (1 - i));/* set half-byte to 10-15 */
-        }
-
-        /* interpret any other character as 0 */
-    }
-
-    return res;
+    return ('0' <= c && c <= '9') ? c - '0'
+         : ('A' <= c && c <= 'F') ? c - 'A' + 10
+         : ('a' <= c && c <= 'f') ? c - 'a' + 10
+         : -1;
 }
 
-size_t ng_netif_addr_from_str(uint8_t *out, size_t out_len, char *str)
+size_t ng_netif_addr_from_str(uint8_t *out, size_t out_len, const char *str)
 {
-    size_t res = 0;
-    char *byte_str = str;
-    int end = 0;
+    if (*str == '\0') {
+        return 0;
+    }
+
+    size_t len = 0;
```

Comment:
[![](http://i1.kym-cdn.com/photos/images/original/000/209/945/D6PfW.jpg)](http://knowyourmeme.com/memes/what-year-is-it)

Seriously, did you learn C in the 80's? Somehow I doubt it. :)